internal/api/ui/login/external_login_handler.go

```diff
@@ -394,6 +407,26 @@ func (l *Login) handleAutoRegister(w http.ResponseWriter, r *http.Request, authR
 	l.renderNextStep(w, r, authReq)
 }
 
+func (l *Login) mapExternalNotFoundOptionFormDataToLoginUser(formData *externalNotFoundOptionFormData) *domain.ExternalUser {
+
+	externalUser := &domain.ExternalUser{
+		IDPConfigID:       formData.externalRegisterFormData.ExternalIDPConfigID,
+		ExternalUserID:    formData.externalRegisterFormData.ExternalIDPExtUserID,
+		PreferredUsername: formData.externalRegisterFormData.Username,
+		DisplayName:       formData.externalRegisterFormData.Email,
+		FirstName:         formData.externalRegisterFormData.Firstname,
+		LastName:          formData.externalRegisterFormData.Lastname,
+		NickName:          formData.externalRegisterFormData.Nickname,
+		Email:             formData.externalRegisterFormData.ExternalEmail,
+		IsEmailVerified:   formData.externalRegisterFormData.ExternalEmailVerified,
+		Phone:			   formData.externalRegisterFormData.ExternalPhone,
+		IsPhoneVerified:   formData.externalRegisterFormData.ExternalPhoneVerified,
+	}
+
+	return externalUser
+
```

Comment:
```suggestion
	isEmailVerified := formData.externalRegisterFormData.ExternalEmailVerified && formData.externalRegisterFormData.Email == formData.externalRegisterFormData.ExternalEmail
	isPhoneVerified := formData.externalRegisterFormData.ExternalPhoneVerified && formData.externalRegisterFormData.Phone == formData.externalRegisterFormData.ExternalPhone
	return &domain.ExternalUser{
		IDPConfigID:       formData.externalRegisterFormData.ExternalIDPConfigID,
		ExternalUserID:    formData.externalRegisterFormData.ExternalIDPExtUserID,
		PreferredUsername: formData.externalRegisterFormData.Username,
		DisplayName:       formData.externalRegisterFormData.Email,
		FirstName:         formData.externalRegisterFormData.Firstname,
		LastName:          formData.externalRegisterFormData.Lastname,
		NickName:          formData.externalRegisterFormData.Nickname,
		Email:             formData.externalRegisterFormData.Email,
		IsEmailVerified:   isEmailVerified,
		Phone:             formData.externalRegisterFormData.Phone,
		IsPhoneVerified:   isPhoneVerified,
	}
```

Reply:
in case the IDP doesn't return an email or phone or the user overwrites them
they will only be verified it they're returned as such from the IDP and are not changed by the user